pkg/cmd/server/start/node_args.go

```diff
@@ -104,13 +99,13 @@ func (args NodeArgs) BuildSerializeableNodeConfig() (*configapi.NodeConfig, erro
 }
 
 // defaultHostname returns the default hostname for this system.
-func defaultHostname() (string, error) {
+func defaultHostname() string {
 
 	// Note: We use exec here instead of os.Hostname() because we
 	// want the FQDN, and this is the easiest way to get it.
-	fqdn, err := exec.Command("hostname", "-f").Output()
+	fqdn, err := exec.Command("uname", "-n").Output()
 	if err != nil {
-		return "", fmt.Errorf("Couldn't determine hostname: %v", err)
+		glog.Fatalf("Couldn't determine hostname: %v", err)
```

Comment:
Even upstream is using `glog.Fatalf` , we should not use it here?
[1] https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/node.go#L31-L34